tools/cli/plugin/data_converter_plugin.go

```diff
@@ -0,0 +1,158 @@
+// The MIT License
+//
+// Copyright (c) 2021 Temporal Technologies Inc.  All rights reserved.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package plugin
+
+import (
+	"fmt"
+	"net/rpc"
+
+	"github.com/hashicorp/go-plugin"
+	commonpb "go.temporal.io/api/common/v1"
+	"go.temporal.io/sdk/converter"
+)
+
+type DataConverterRPC struct {
+	client *rpc.Client
+}
+
+func NewDataConverterPlugin(name string) (converter.DataConverter, error) {
+	client, err := newPluginClient(DATA_CONVERTER_PLUGIN_TYPE, "tctl-plugin-dataconverter-"+name)
+	if err != nil {
+		return nil, fmt.Errorf("unable to register plugin: %v\n", err)
+	}
+
+	dataConverter, ok := client.(converter.DataConverter)
+	if !ok {
+		return nil, fmt.Errorf("incorrect plugin type")
+	}
+
+	return dataConverter, nil
+}
+
+func (g *DataConverterRPC) FromPayload(payload *commonpb.Payload, valuePtr interface{}) error {
+	err := g.client.Call("Plugin.FromPayload", payload, valuePtr)
```

Comment:
`"Plugin.FromPayload"` is essentially a late binding. I believe unit test coverage is required here. What if interface methods are renamed. It is also hard to test all this stuff manually. Does hashi provide any facilities for unit tests?